infertrade/algos/community/signals.py

```diff
@@ -50,6 +51,47 @@ def high_low_diff_scaled(df: pd.DataFrame, amplitude: float = 1) -> pd.DataFrame
     df["signal"] = (df["high"] - max(df["low"])) * amplitude
     return df
 
+def simple_moving_average(df: pd.DataFrame, window: int = 3) -> pd.DataFrame:
+    """
+    Calculates smooth signal based on price trends by filtering out the noise from random short-term price fluctuations
+    """
+    df['signal'] = df["close"].rolling(window=window).mean()
+    return df
+
+def weighted_moving_average(df: pd.DataFrame, window: int = 3) -> pd.DataFrame:
+    """
+    Weighted moving averages assign a heavier weighting to more current data points since they are more relevant than data points in the distant past. 
+    """
+    weights = np.arange(1, window+1)
+    weights /= weights.sum()
+    df["signal"] = df["close"].rolling(window=window).apply(lambda a: a.mul(weights).sum())
+    return df
+
+def exponentially_weighted_moving_average(df: pd.DataFrame, window: int = 3) -> pd.DataFrame:
+    """
+    This function uses an exponentially weighted multiplier to give more weight to recent prices.
+    """
+    df["signal"] = df["close"].ewm(span=window, adjust=False).mean()
+    return df
+
+def moving_average_convergence_divergence(df: pd.DataFrame, lower_window: int =12, upper_window: int = 26) -> pd.DataFrame:
+    """
+    This function is a trend-following momentum indicator that shows the relationship between two moving averages at different windows:
+    The MACD is usually calculated by subtracting the 26-period exponential moving average (EMA) from the 12-period EMA.
+    
+    """
+    ewma_26 = exponentially_weighted_moving_average(df , window = lower_window)
+    ewma_12 = exponentially_weighted_moving_average(df, window= upper_window)
+
+    # MACD calculation
+    macd = ewma_26["signal"]-ewma_12["signal"]
+
+    # convert MACD into signal
+    df["signal"]= macd.ewm(span=9, adjust=False).mean()
+
+
+    return df
```

Comment:
MACD is one of the functions in the ta library, so you could check you get the same results?

Reply:
Also for signals to be accessible via the API we need to add details to the dictionary at the end of the file (same with allocations).